Framework/DataObjects/src/PeaksWorkspace.cpp

```diff
@@ -149,6 +150,28 @@ void PeaksWorkspace::removePeak(const int peakNum) {
   peaks.erase(peaks.begin() + peakNum);
 }
 
+/** Removes multiple peaks
+* @param badPeaks peaks to be removed
+*/
+void PeaksWorkspace::removePeaks(std::vector<int> &badPeaks) {
```

Comment:
And change this to 
`(std::vector<int> &&badPeaks)`

And the callers to
`removePeaks(std::move(myVec));`

This way we take ownership of the container without making a copy.